university-agent/src/test/java/nl/quintor/studybits/controller/LoginIT.java

```diff
@@ -69,134 +89,142 @@ public static void bootstrapBackend() throws Exception {
-        credentialOffer.setTheirDid(credentialOfferEnvelopes.getDid());
-        assertThat(credentialOffer.getSchemaId(), notNullValue());
+        // Student logs in to university with correct password
+        MessageEnvelope<ConnectionResponse> connectionResponseMessageEnvelope = givenCorrectHeaders(ENDPOINT_RUG)
+                .queryParam("student_id", "12345678")
+                .queryParam("password", "test1234")
+                .body(studentCodec.encryptMessage(connectionRequest, CONNECTION_REQUEST, rugVerinymDid).get().toJSON())
+                .post("/agent/login")
+                .then()
+                .assertThat().statusCode(200)
+                .extract().as(MessageEnvelope.class);
 
-        Prover prover = new Prover(studentWallet, "master_secret_name");
-        prover.init();
+        ConnectionResponse connectionResponse = studentCodec.decryptMessage(connectionResponseMessageEnvelope).get();
 
-        CredentialRequest credentialRequest = prover.createCredentialRequest(credentialOffers.getCredentialOffers().get(0)).get();
+        //New DID created by student
+        log.debug("Lisa RUG DID: " + connectionResponse.getDid());
+        rugLisaDid = connectionResponse.getDid();
+        // Decrypt and accept connection response
+        studentWallet.acceptConnectionResponse(connectionResponse, connectionResponseMessageEnvelope.getDid()).get();
+    }
 
-        MessageEnvelope authcryptedCredentialRequestEnvelope = studentCodec.encryptMessage(credentialRequest, IndyMessageTypes.CREDENTIAL_REQUEST).get();
+    @Test
+    public void test3_GetCredentialOffers() throws IndyException, ExecutionException, InterruptedException, IOException {
+        //Request CREDENTIAL_OFFERS
+        String getRequest = studentCodec.encryptMessage(CREDENTIAL_OFFERS.getURN(), GET_REQUEST, rugLisaDid).get().toJSON();
 
-        MessageEnvelope<CredentialWithRequest> credentialEnvelope = givenCorrectHeaders(ENDPOINT_RUG)
-                .body(authcryptedCredentialRequestEnvelope)
+        MessageEnvelope<CredentialOfferList> credentialOfferEnvelopes = givenCorrectHeaders(ENDPOINT_RUG)
+                .body(getRequest)
                 .post("/agent/message")
                 .then()
                 .assertThat().statusCode(200)
                 .extract().as(MessageEnvelope.class);
 
-        assertThat(credentialEnvelope.getMessageType().getURN(), is(equalTo(IndyMessageTypes.CREDENTIAL.getURN())));
-
-        CredentialWithRequest credentialWithRequest = studentCodec.decryptMessage(credentialEnvelope).get();
+        assertThat(credentialOfferEnvelopes.getMessageType(), is(CREDENTIAL_OFFERS));
 
-        prover.storeCredential(credentialWithRequest).get();
+        studentCredentialOfferList = studentCodec.decryptMessage(credentialOfferEnvelopes).get();
 
-        Credential credential = credentialWithRequest.getCredential();
+        assertThat(studentCredentialOfferList.getCredentialOffers().isEmpty(), is(false));
```

Comment:
Let's check that the size is 1, save just the one offer and remove the loop in the next test.